File: app/src/debug/java/com/example/trackers/FrescoCacheStatsTracker.java
package com.example.trackers;

import com.example.android.utils.LogUtility;
import com.facebook.cache.common.CacheKey;
import com.facebook.imagepipeline.cache.CountingMemoryCache;
import com.facebook.imagepipeline.cache.ImageCacheStatsTracker;

public class FrescoCacheStatsTracker implements ImageCacheStatsTracker {
	@Override
	public void onBitmapCachePut() {

	}

	@Override
	public void onBitmapCacheHit(CacheKey cacheKey) {
		LogUtility.d("[ Fresco Bitmap Cache Hit ] " + cacheKey);
	}

	@Override
	public void onBitmapCacheMiss() {

	}

	@Override
	public void onMemoryCachePut() {

	}

	@Override
	public void onMemoryCacheHit(CacheKey cacheKey) {
		LogUtility.d("[ Fresco Memory Cache Hit ] " + cacheKey);
	}

	@Override
	public void onMemoryCacheMiss() {

	}

	@Override
	public void onStagingAreaHit(CacheKey cacheKey) {
		LogUtility.d("[ Fresco Staging Area Hit ] " + cacheKey);

	}

	@Override
	public void onStagingAreaMiss() {

	}

	@Override
	public void onDiskCacheHit() {
		LogUtility.d("[ Fresco Disk Cache Hit ]");
	}

	@Override
	public void onDiskCacheMiss() {

	}

	@Override
	public void onDiskCacheGetFail() {

	}

	@Override
	public void registerBitmapMemoryCache(CountingMemoryCache<?, ?> bitmapMemoryCache) {

	}

	@Override
	public void registerEncodedMemoryCache(CountingMemoryCache<?, ?> encodedMemoryCache) {

	}
}

File: app/src/debug/kotlin/com/example/trackers/FrescoCacheStatsTracker.kt
package com.example.trackers

import com.example.android.utils.LogUtility
import com.facebook.cache.common.CacheKey
import com.facebook.imagepipeline.cache.CountingMemoryCache
import com.facebook.imagepipeline.cache.ImageCacheStatsTracker

class FrescoCacheStatsTracker : ImageCacheStatsTracker {
  override fun onBitmapCachePut() {

  }

  override fun onBitmapCacheHit(cacheKey: CacheKey) {
    LogUtility.d("[ Fresco Bitmap Cache Hit ] $cacheKey")
  }

  override fun onBitmapCacheMiss() {

  }

  override fun onMemoryCachePut() {

  }

  override fun onMemoryCacheHit(cacheKey: CacheKey) {
    LogUtility.d("[ Fresco Memory Cache Hit ] $cacheKey")
  }

  override fun onMemoryCacheMiss() {

  }

  override fun onStagingAreaHit(cacheKey: CacheKey) {
    LogUtility.d("[ Fresco Staging Area Hit ] $cacheKey")

  }

  override fun onStagingAreaMiss() {

  }

  override fun onDiskCacheHit() {
    LogUtility.d("[ Fresco Disk Cache Hit ]")
  }

  override fun onDiskCacheMiss() {

  }

  override fun onDiskCacheGetFail() {

  }

  override fun registerBitmapMemoryCache(bitmapMemoryCache: CountingMemoryCache<*, *>) {

  }

  override fun registerEncodedMemoryCache(encodedMemoryCache: CountingMemoryCache<*, *>) {

  }
}
